### DISCUSSION TITLE
deselect event firing after delete event

### DISCUSSION THREAD

**Discussion by markwatabe** [xyflow/xyflow#2036] [Q&A]
Maybe this is intentional, but it seems strange to me: If I select a node and delete, two separate events fire in this order:

1.  { id: "someNode", type: "remove"}
2. { id: "someNode", selected: false, type: "select" }

Do you need the de-select event if that node is deleted anyways? If you do, I would think it should fire before the delete event, because semantically it makes no sense to fire a deselect event for a node which is no longer meant to exist.

**Comment by markwatabe** [comment]
The more I play around with the API for V10, the more challenging I find it compared to V9. Curious if anyone has similar thoughts. Here is an example of what I find cumbersome:

Suppose I select a node (N) with two edges (A & B), and I select one of its edges (A) as well, and then press delete. I get the following events:
- Edge update with three items:
  - {id: A, type: 'remove'}
  - {id: A, type: 'remove'} (twice because it was deleted by being selected and because it was connected to a deleted node)
  - {id: B, type: 'remove'}
- Node update with one item { id: "N", type: "remove" }
- Node update with one item  { id: "N", type: "select", selected: false }
- Edge update with one item { id: "A", type: "select", selected: false }

I would think that is it, but then I also get:
- Edge update with two items:
  - {id: A, type: 'remove'}
  - {id: B, type: 'remove'}
- Node update with one item { id: "N", type: "remove" }
- Node update with one item  { id: "N", type: "select", selected: false }

I suppose I could treat those events similar to how I was doing event handling before, but these don't seem to be batched well at all, and it will require a lot of special logic to get right.

**Reply by moklick** [reply]
Thanks for the report. We can do some cleanup for the remove events! Normally you wouldn't need to care about these change events but use the `applyNodeChanges`/`applyEdgeChanges`. We still have the onNodesDelete/onEdgesDelete callbacks btw if you are looking for that.

**Reply by moklick** [reply]
Hey @markwatabe 

could you try v10.0.8. I did some cleanup for the change events.

**Reply by markwatabe** [reply]
Thanks for doing that!

I haven't had time to look into it yet, but just wanted to drop a quick note that I'm still hoping to migrate to V10, and become a PRO supporter soon. Thanks for the great package.
